### PR TITLE
Remove nfcore_external_java_deps.jar from lib directory in pipeline template

### PR DESCRIPTION
**Pull request by drpatelh**
This is no longer required as it should have been replaced by the nf-validation plugin.

### CODE REVIEW

**Comment by drpatelh** [comment]
Tests are failing due to:

```
╭─ [✗] 2 Pipeline Tests Failed ────────────────────────────────────────────────╮
│                                                                              │
│ files_exist: File not found: lib/nfcore_external_java_deps.jar               │
│ files_unchanged: File does not exist: lib/nfcore_external_java_deps.jar      │
│                                                                              │
╰──────────────────────────────────────────────────────────────────────────────╯
```

Do we generate a lint error in the future if this file is found in the template or we keep it there for those to remove if they are using `nf-validation`?

File: nf_core/lint/files_exist.py [hunk 1]
```diff
@@ -235,7 +240,23 @@ def pf(file_path):
...
+            continue
+        nextflow_config = pf("nextflow.config")
+        in_config = False
+        with open(nextflow_config) as f:
+            if file[1] in f.read():
+                in_config = True
+        if os.path.isfile(pf(file[0])) and in_config:
+            failed.append(f"File must be removed: {self._wrap_quotes(file[0])}")
+        elif os.path.isfile(pf(file[0])) and not in_config:
+            passed.append(f"File not found check: {self._wrap_quotes(file[0])}")
```
Comment (mirpedrol) [inline nf_core/lint/files_exist.py:255]: This message confused me a little bit, would it be better to remove the `not`?
```suggestion
            passed.append(f"File found check: {self._wrap_quotes(file[0])}")
```

File: nf_core/lint/files_unchanged.py [hunk 2]
```diff
@@ -208,6 +214,40 @@ def _tf(file_path):
...
+            ignored.append(f"File ignored due to lint config: {self._wrap_quotes(files)}")
+
+        # Ignore if we can't find the file
+        elif not any([_pf(files[0]).is_file()]):
+            ignored.append(f"File does not exist: {self._wrap_quotes(files[0])}")
+
+        # Check that the file has an identical match
+        else:
+            config_key, config_value = list(files[1].items())[0]
+            if config_key in self.nf_config and self.nf_config[config_key] == config_value:
```
[the remark this answers is not in the document]
Comment (mashehu) [reply nf_core/lint/files_unchanged.py:231]: thanks! handle that case now  as well

**Review by mirpedrol — APPROVED** [review]
LGTM!

**Pull request merged**